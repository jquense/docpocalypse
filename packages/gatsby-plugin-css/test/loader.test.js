const { webpack } = require('./helpers');

describe('Loader', () => {
  test('Default', () => {
    const config = {
      loader: {
        test: /\.css$/,
        options: {
          plugins: [(css) => css],
        },
      },
    };

    return webpack('css/index.js', config).then((stats) => {
      const { source } = stats.toJson().modules[1];

      expect(source).toEqual('module.exports = "a { color: black }\\n"');
      expect(source).toMatchSnapshot();
    });
  });

  test('uses previous AST', () => {
    const spy = jest.fn();
    const config = {
      rules: [
        {
          test: /style\.js$/,
          use: [
            {
              loader: require.resolve('../postcss-loader'),
              options: { importLoaders: 1 },
            },
            {
              loader: require.resolve('./ast-loader'),
              options: { spy },
            },
          ],
        },
      ],
    };

    return webpack('jss/index.js', config).then((stats) => {
      const { source } = stats.toJson().modules[1];

      expect(spy).toHaveBeenCalledTimes(1);
      expect(source).toMatchSnapshot();
    });
  });
});
