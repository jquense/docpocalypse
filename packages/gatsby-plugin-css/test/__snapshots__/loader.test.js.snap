// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Default 1`] = `"module.exports = \\"a { color: black }\\\\n\\""`;

exports[`Loader uses previous AST 1`] = `"module.exports = \\"a {\\\\n    color: yellow\\\\n}\\""`;
